Mathlib/Combinatorics/SimpleGraph/Metric.lean

```diff
@@ -122,6 +122,24 @@ theorem dist_comm {u v : V} : G.dist u v = G.dist v u := by
     simp [h, h', dist_eq_zero_of_not_reachable]
 #align simple_graph.dist_comm SimpleGraph.dist_comm
 
+/- The distance between vertices is equal to `1` if and only if these vertices are adjacent. -/
+theorem dist_eq_one_iff_adj {u v : V} : G.dist u v = 1 ↔ G.Adj u v := by
+  refine ⟨fun h ↦ ?_, fun h ↦ ?_⟩
+  · have : u ≠ v ∧ G.Reachable u v := by
+      rw [and_iff_not_or_not, ne_eq, not_not, ← Iff.not dist_eq_zero_iff_eq_or_not_reachable, h]
+      apply one_ne_zero
+    obtain ⟨w, hw⟩ := Reachable.exists_walk_of_dist this.2
```

Comment:
Please add `Reachable.of_dist_ne_zero` and `exists_walk_of_dist_ne_zero`.

Reply:
Good idea, working on it!